src/query/service/src/interpreters/interpreter_copy_v2.rs

```diff
@@ -113,6 +113,26 @@ impl CopyInterpreterV2 {
         }
     }
 
+    async fn purge_files(&self, from: &ReadDataSourcePlan, files: &Vec<String>) -> Result<()> {
+        match &from.source_info {
+            SourceInfo::StageSource(table_info) => {
+                if table_info.stage_info.copy_options.purge {
+                    let rename_me: Arc<dyn TableContext> = self.ctx.clone();
+                    let op = StageSourceHelper::get_op(&rename_me, &table_info.stage_info).await?;
+                    for path in files {
+                        op.object(path).delete().await?;
```

Comment:
Suggest not returning an error if the `delete` opeartion failed. We should skip it and write a log.